quality_control_stock_oca/models/stock_picking.py

```diff
@@ -80,5 +80,25 @@ def _action_done(self):
                     )
                 )
             for trigger_line in _filter_trigger_lines(trigger_lines):
-                inspection_model._make_inspection(operation, trigger_line)
+                inspection = inspection_model._make_inspection(operation, trigger_line)
+                if operation.product_id.tracking != "none":
+                    move_line = operation.move_line_ids[0]
+                    if not trigger_line.trigger.inspection_per_lot:
+                        inspection.write(
+                            {
+                                "lot_id": move_line.lot_id and move_line.lot_id.id,
+                            }
+                        )
```

Comment:
Can  a operation have a first move line without lot and a second move line with lot?

Maybe it is better to write: {"lot_id": operation.mapped("move_line_ids.lot_id.id")[:1]}

Reply:
I think that all stock move lines related to products with tracking need a serial/lot number when validating the picking. This behaviour has not actually being modified from the previous version of the module.